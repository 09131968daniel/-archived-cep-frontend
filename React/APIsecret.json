{"web":{"client_id":"735505547798-12t2nvnqjii71tipepkrjcpt05h4h2c1.apps.googleusercontent.com","project_id":"bookoutlet","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"OS0H2o8_0mpuNc8YTh9b5IZ1"}}