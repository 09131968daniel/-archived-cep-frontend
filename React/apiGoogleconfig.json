{
  "clientId": "113121693676-ea55mom4ehob108vt02gvqgnq5nnun03.apps.googleusercontent.com",
  "apiKey": "AIzaSyCfkKH3QjGrw_LW4-NpTrIER-kb_KpU0qQ",
  "scope": "https://www.googleapis.com/auth/calendar",
  "discoveryDocs": [
    "https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest"
  ]
}
