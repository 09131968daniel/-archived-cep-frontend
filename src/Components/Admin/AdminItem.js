import React, { useState, useEffect } from "react";
import {
  Grid,
  Typography,
  makeStyles,
  Card,
  CardContent,
  CardHeader,
  Button,
} from "@material-ui/core";
import Axios from "axios";
import { useSelector } from "react-redux";

//used solely for styling
const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
  },
  left: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "flex-start",
  },
  right: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center",
  },
  middle: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
  },
  rightButton: {
    marginTop: "5px",
    marginBottom: "5px",
  },
  spacing: {
    marginTop: "15px",
    marginBottom: "15px",
  },
}));

//Display individual requests
export default function AdminItem(props) {
  const token = useSelector((state) => state.credReducer.token);
  const styles = useStyles();

  //state to hold user object
  const [userData, setUserData] = useState({});

  //Status=Pending color is Orange, Status=Completed color is Blue
  //Initial color is Orange
  const [statusColor, setStatusColor] = useState("#F26925");
  const [status, setStatus] = useState(props.data.status);

  //States used for conditional rendering
  const [buttonCompleteVisi, setButtonCompleteVisi] = useState(true);
  const [buttonDeleteVisi] = useState(true);
  const [cardVisi, setCardVisi] = useState(true);

  //Simply return the button component for conditional rendering
  const ButtonComplete = () => {
    return (
      <Button
        className={styles.rightButton}
        variant="outlined"
        color="primary"
        onClick={handleToComplete}
      >
        Completed
      </Button>
    );
  };
  const ButtonDelete = () => {
    return (
      <Button
        className={styles.rightButton}
        variant="outlined"
        color="secondary"
        onClick={handleDelete}
      >
        Delete
      </Button>
    );
  };

  //Card that will display the user info and request info
  const CardInfo = () => {
    return (
      <Card className={styles.spacing}>
        <CardHeader style={{ backgroundColor: statusColor }}></CardHeader>
        <CardContent>
          <Grid container spacing={3}>
            <Grid item xs={3} className={styles.left}>
              <Typography variant="overline">{userData.company}</Typography>
              <Typography variant="h4">
                {userData.firstName} {userData.lastName}
              </Typography>
              <Typography variant="h6">{userData.email}</Typography>
              <Typography variant="h6">{userData.phone}</Typography>
              <Typography variant="h6">{props.data.requestType}</Typography>
            </Grid>
            <Grid item xs={6} className={styles.middle}>
              <Typography variant="h4">{props.data.batchId}</Typography>
              <Typography variant="body2">{props.data.description}</Typography>
            </Grid>
            <Grid item xs={3} className={styles.right}>
              <Typography variant="body2">{props.data.startTime}</Typography>
              {buttonDeleteVisi ? <ButtonDelete /> : null}
              {buttonCompleteVisi ? <ButtonComplete /> : null}
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    );
  };

  function changeUserData(data) {
    setUserData(data);
  }

  //On first render check if the status is complete and render the correct color and buttons
  //Make a axios call to get the user data that is mapped to a particular request by finding it by userId
  useEffect(() => {
    if (status === "Done") {
      setStatusColor("#72A4C2");
      setButtonCompleteVisi(false);
    }
    var axios = require("axios");

    var config = {
      method: "get",
      url: `http://ec2-18-232-171-89.compute-1.amazonaws.com:8081/users/user/?id=${props.data.userId}`,
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };

    axios(config)
      .then(function (response) {
        changeUserData(response.data);
      })
      .catch(function (error) {
        console.log(error);
      });

    //This does the same thing as the code from lines 135-151, but that code was simply imported from postman
    // Axios.get(`http://ec2-18-232-171-89.compute-1.amazonaws.com:8081/users/user/?id=${props.data.userId}`)
    // .then((response)=>{
    //     changeUserData(response.data);
    // })
    // .catch((err) => console.log());
  }, []);

  /* handle complete button to change to the color, status,
  and call a function to persist the complete status of the request */
  const handleToComplete = () => {
    setStatus("Done");
    setStatusColor("#72A4C2");
    setButtonCompleteVisi(false);
    updateToComplete();
  };

  //Persists Complete of request to database
  const updateToComplete = () => {
    //axios call
    Axios({
      method: "put",
      url: `http://ec2-18-232-171-89.compute-1.amazonaws.com:8081/users/admin/request/update/${props.data.requestId}`,
      headers: {
        Authorization: `Bearer ${token}`,
      },
      data: {
        status: "Done",
      },
    })
      .then((response) => console.log("Success"))
      .catch((err) => console.log("Failure"));
  };

  //Handle onClick delete
  /*TODO: Do not delete the actual records. This was super early on to clean up before "Status" existed in the request table. 
  Now just update the status to "Archive" and have the back-end get all records that do not have the "Archieve" status */
  const handleDelete = () => {
    Axios.delete(
      `http://ec2-18-232-171-89.compute-1.amazonaws.com:8081/users/admin/request/delete/${props.data.requestId}`,
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      },
    )
      .then((response) => {
        //Simply conditionally hide the card to simulate archive
        setCardVisi(false);
      })
      .catch((err) => console.log("Failure"));
  };

  return <div>{cardVisi ? <CardInfo /> : null}</div>;
}
