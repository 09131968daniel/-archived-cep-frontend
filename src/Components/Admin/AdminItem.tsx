import React, { useState, useEffect, ReactElement, SyntheticEvent } from "react";
import {
  Grid,
  Typography,
  makeStyles,
  Card,
  CardContent,
  CardHeader,
  Button,
  StyleRules,
} from "@material-ui/core";
import Axios from "axios";
import { useSelector } from "react-redux";
import moment from 'moment';

//used solely for styling
const useStyles: Function = makeStyles((): StyleRules => ({
  spacing: {
    marginTop: "15px",
    marginBottom: "15px",
    height: "100%"
  },
  button: {
    margin: "0 5px"
  }
}));

//Display individual requests
export default function AdminItem(props: any): ReactElement {

  interface styleINF {
    spacing: string,
    button: string
  }

  const token: String = useSelector((state: any) => state.credReducer.token);
  const styles: styleINF = useStyles();

  //Userstate
  const [userData, setUserData] = useState({
    company: String,
    firstName: String,
    lastName: String,
    email: String,
    phone: String
  });

  //Status=Pending color is Orange, Status=Completed color is Blue
  //Initial color is Orange
  const [statusColor, setStatusColor] = useState("#F26925");
  const [status, setStatus] = useState(props.data.status);

  //States used for conditional rendering
  const [buttonCompleteVisi, setButtonCompleteVisi] = useState(true);
  const [buttonDeleteVisi] = useState(true);
  const [cardVisi, setCardVisi] = useState(true);

  //Simply return the button component for conditional rendering
  const ButtonComplete: React.FC = (): ReactElement => {
    return (
      <Button
        variant="outlined"
        color="primary"
        onClick={handleToComplete}
        className={styles.button}
      >
        Mark as complete
      </Button>
    );
  };
  const ButtonDelete: React.FC = (): ReactElement => {
    return (
      <Button
        variant="outlined"
        color="secondary"
        onClick={handleDelete}
        className={styles.button}
      >
        Cancel/Delete
      </Button>
    );
  };

  const CardInfo: React.FC = (): ReactElement => {
    return (
      <Card className={styles.spacing}>
        <CardHeader style={{ backgroundColor: statusColor }} title={userData.company}></CardHeader>
        <CardContent>
          <p>Made By: {userData.firstName} {userData.lastName}</p>
          <p>Email: {userData.email}</p>
          <p>For: {props.data.requestType}</p>
          <p>{(props.data.batchId) ? "Batch: " + props.data.batchId : "No Batch Specified"}</p>
          <p>Description: {props.data.description}</p>
          
          <p>Date: {moment(props.data.startTime).format("MM/DD/YYYY hh:mm a")}</p>

          {buttonCompleteVisi ? <ButtonComplete /> : null}
          {buttonDeleteVisi ? <ButtonDelete /> : null}

        </CardContent>
      </Card>
    );
  };

  // function changeUserData(data:Object):void {
  function changeUserData(data: any): void {
    setUserData(data);
  }
  //On first render check if the status is complete and render the correct color and buttons
  useEffect((): void => {
    if (status === "Done") {
      setStatusColor("#72A4C2");
      setButtonCompleteVisi(false);
    }
    const axios = require("axios");
    const config = {
      method: "get",
      url: `${process.env.REACT_APP_ZUUL_ROUTE}/users/user/?id=${props.data.userId}`,
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };

    axios(config)
      .then(function (response: any) {
        changeUserData(response.data);
      })
      .catch(function (error: any) {
        console.log(error);
      });
  }, []);

  //handle complete button to change to the color, status,
  //and call a function to persist the complete status of the request
  // const handleToComplete:Function = (event:Event):void => {
  function handleToComplete(event: SyntheticEvent): void {
    setStatus("Done");
    setStatusColor("#72A4C2");
    setButtonCompleteVisi(false);
    updateToComplete();
  };

  //Persists Complete of request to database
  const updateToComplete: Function = (): void => {
    //Get JWT

    //axios call
    Axios({
      method: "put",
      url: `${process.env.REACT_APP_ZUUL_ROUTE}/users/admin/request/update/${props.data.requestId}`,
      headers: {
        Authorization: `Bearer ${token}`,
      },
      data: {
        status: "Done",
      },
    })
      .then((response) => console.log("Success"))
      .catch((err) => console.log("Failure"));
  };

  //Handle onClick delete
  // const handleDelete:Function = (event:Event):void => {
  function handleDelete(event: SyntheticEvent): void {
    Axios.delete(
      `${process.env.REACT_APP_ZUUL_ROUTE}/users/admin/request/delete/${props.data.requestId}`,
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      },
    )
      .then(() => {
        /*After delete we reload the webpage so it can show "real time" that the request has been deleted
            May want to just hide the card to not lose potential filtering and sorting options later on*/
        //window.location.reload();
        setCardVisi(false);
      })
      .catch((err) => console.log("Failure"));
  };

  //These props need to change to match the data that is given
  //Change "requestType" to "technology" and "endTime" to "date" if you want to mock test with the ideal/test look of the request
  return <React.Fragment>{cardVisi ? <CardInfo /> : null}</React.Fragment>;
}
