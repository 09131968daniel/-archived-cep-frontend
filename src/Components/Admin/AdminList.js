import React, { useState, useEffect } from "react";
import AdminItem from "./AdminItem";
import axios from "axios";
import { useSelector } from "react-redux";

//Show a list of requests
export default function AdminList() {
  const token = useSelector((state) => state.credReducer.token);
  const [realData, setRealData] = useState([]);

  function changeRealData(data) {
    setRealData(data);
  }

  //Make an axios call to display the list of requests
  useEffect(() => {
    axios
      .get(
        "http://ec2-18-232-171-89.compute-1.amazonaws.com:8081/interventions",
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      )
      //  .then(res=>res.json())
      .then((response) => {
        //Console.log used to check the fields to set up the adminItem for later
        console.log(response.data);
        changeRealData(response.data);
      })
      .catch((err) => console.log());
  }, []);

  //Render the list of requests
  //Change "realData" to "testData" if you want to mock test with the ideal/test look of the request
  return (
    <div>
      {realData.map((data) => {
        return <AdminItem data={data} key={data.requestId} />;
      })}
    </div>
  );
}
