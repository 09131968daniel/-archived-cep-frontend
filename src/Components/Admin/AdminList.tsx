import React, { useState, useEffect } from "react";
import AdminItem from "./AdminItem";
import axios from "axios";
import { useSelector } from "react-redux";
import apiBasePath from "../../apiBasePath";

//Show a list of requests
export default function AdminList() {
  const token = useSelector((state: any) => state.credReducer.token);
  const [realData, setRealData] = useState([]);

  function changeRealData(data: any) {
    setRealData(data);
  }

  //Make an axios call to display the list of requests
  useEffect(() => {
    axios
      .get(
        apiBasePath + "/interventions",
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      )
      //  .then(res=>res.json())
      .then((response) => {
        //Console.log used to check the fields to set up the adminItem for later
        console.log(response.data);
        changeRealData(response.data);
      })
      .catch((err) => console.log());
  }, []);

  //Render the list of requests
  //Change "realData" to "testData" if you want to mock test with the ideal/test look of the request
  return (
    <div>
      {realData.map((data: any) => {
        return <AdminItem data={data} key={data.requestId} />;
      })}
    </div>
  );
}
