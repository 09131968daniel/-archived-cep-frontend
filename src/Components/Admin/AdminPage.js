import React from 'react'
import AdminList from './AdminList'
import { makeStyles, Container } from '@material-ui/core'

const useStyles = makeStyles(() => ({
    root: {
        backgroundColor: 'Aqua'
    },
}));

//Displays components for Admin
export default function AdminPage(){
    const styles = useStyles();
    
    return(
        <div>
            <Container>
                <AdminList/>
            </Container>
        </div>
    )
}