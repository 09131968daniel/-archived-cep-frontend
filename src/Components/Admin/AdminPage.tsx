import React, { useEffect, ReactElement } from 'react'
import AdminList from './AdminList'
import { makeStyles, Container } from '@material-ui/core'
import { useDispatch } from 'react-redux';
import { dispatchLink } from '../../redux/actions/redirectAction'

const useStyles:Function = makeStyles(():any => ({
    root: {
        backgroundColor: 'Aqua'
    },
}));
//Displays components for Admin
//Might might not even need this component
export default function AdminPage():ReactElement{
    const styles:Object = useStyles();
    const dispatch:Function = useDispatch();

    useEffect(():void => {
        dispatch(dispatchLink('/admin'));
    }, [])

    return(
        <div>
            <Container>
                <AdminList/>
            </Container>
        </div>
    )
}