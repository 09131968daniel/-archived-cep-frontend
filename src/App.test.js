import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminItem from './Components/Admin/AdminItem';
import AdminList from './Components/Admin/AdminList';
import AdminPage from './Components/Admin/AdminPage';

Enzyme.configure({ adapter: new Adapter() });

describe('AdminItem', ()=>{
  it('should render correctly in "debug" mode', ()=>{
    const props = {
      data: {
        companyName: "Test",
        firstName: "Johnny",
        lastName: "Test",
        technology: "Java-React",
        date: "Jan 11, 2020",
        descript: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        status: "Pending"
      }
    }
    const component = Enzyme.shallow(<AdminItem {...props} />);

    component.setProps({stats:"Complete"})
    expect(component).toMatchSnapshot();
  });
});

describe('AdminList', ()=>{
  it('should render correctly in "debug" mode', ()=>{
    const component = Enzyme.shallow(<AdminList debug />);

    expect(component).toMatchSnapshot();
  });
});

describe('AdminPage', ()=>{
  it('should render correctly in "debug" mode', ()=>{
    const component = Enzyme.shallow(<AdminPage debug />);

    expect(component).toMatchSnapshot();
  });
});

