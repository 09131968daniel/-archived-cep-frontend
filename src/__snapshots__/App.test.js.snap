// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminItem should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`AdminList should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`AdminPage should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
